pkg/daemon/ceph/client/pool.go

```diff
@@ -179,7 +179,16 @@ func ParsePoolDetails(in []byte) (CephStoragePoolDetails, error) {
 }
 
 func CreatePool(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, pool cephv1.NamedPoolSpec, appName string) error {
-	return CreatePoolWithPGs(context, clusterInfo, clusterSpec, pool, appName, DefaultPGCount)
+	var pgCount string
+	if pool.PGs > 0 {
+		pgCount = strconv.Itoa(pool.PGs)
+	} else {
+		pgCount = DefaultPGCount
+	}
+	if pool.AutoscaleMode == "" {
+		pool.AutoscaleMode = DefaultAutoscaleMode
```

Comment:
We prefer not to modify the values in the CRD types, and treat them as read-only if possible, to preserve the user's original desired state. We only have a few exceptions where it made the implementation a lot simpler. So perhaps we just set the default inside the methods where we actually use the `"--autoscale-mode"` param, since this is simple enough.